pkg/transformers/builder/transformer_builder.go

```diff
@@ -3,46 +3,115 @@
 package builder
 
 import (
+	"fmt"
+
 	"github.com/xataio/pgstream/pkg/transformers"
 	"github.com/xataio/pgstream/pkg/transformers/greenmask"
 	"github.com/xataio/pgstream/pkg/transformers/neosync"
 )
 
+// ErrUnknownParameter is returned when unknown parameter is provided
+var ErrUnknownParameter = fmt.Errorf("unknown parameter provided to transformer")
+
 func New(cfg *transformers.Config) (transformers.Transformer, error) {
 	switch cfg.Name {
 	case transformers.GreenmaskString:
+		if err := validateParameters(cfg.Parameters, greenmask.StringTransformerParams); err != nil {
```

Comment:
Would it make sense to have the validation done within the transformer constructor, rather than exposing all the transformer parameters, which would otherwise be internal?
The function can be defined in the transformers package (like the other transformer parameter util functions), but be called within the constructor, which is the one that should make sure the input parameters are compatible with the implementation. 

Otherwise, imagine a situation where instead of using this builder, the constructor is called directly, and the validation doesn't happen.